Use email address from Commitments rather than simulated event

benagain opened the pull request.

Pauljgraham reviewed (changes requested):
Not all apprenticeship records will have an email on them. So you need to ignore those without an apprenticeship Id

benagain commented:
> 
> 
> Not all apprenticeship records will have an email on them. So you need to ignore those without an apprenticeship Id

Yes, you're right.  Because we've moved from our simulated event (that we only ever receive for apprenticeships we care about) to the real event (that we will receive for all apprenticeships, even ones that we should ignore).

I've added a check.
